docs/tutorial/category_theory/calculating_colimits_in_Top.lean

```diff
@@ -0,0 +1,94 @@
+import category_theory.instances.Top
+import category_theory.limits.shapes
+import topology.instances.real
+
+/- This file contains some demos of using the (co)limits API to do topology. -/
+
+noncomputable theory
+
+open category_theory
+open category_theory.instances
+open category_theory.limits
+
+def R : Top := Top.of ℝ
+def I : Top := Top.of (set.Icc 0 1 : set ℝ)
+def pt : Top := Top.of unit
+
+section MappingCylinder
+-- Let's construct the mapping cylinder.
+def to_pt (X : Top) : X ⟶ pt :=
+{ val := λ _, unit.star, property := continuous_const }
+def I_0 : pt ⟶ I :=
+{ val := λ _, ⟨(0 : ℝ), begin simp, split; norm_num, end⟩,
+  property := continuous_const }
+def I_1 : pt ⟶ I :=
+{ val := λ _, ⟨(1 : ℝ), begin simp, split; norm_num, end⟩,
+  property := continuous_const }
+
+def cylinder (X : Top) : Top := limit (pair X I)
+-- To define a map to the cylinder, we give a map to each factor.
+-- `binary_fan` is a helper method for constructing a `cone` over `pair X Y`.
+def cylinder_0 (X : Top) : X ⟶ cylinder X :=
+limit.lift (pair X I) (binary_fan (𝟙 X) (to_pt X ≫ I_0))
+def cylinder_1 (X : Top) : X ⟶ cylinder X :=
+limit.lift (pair X I) (binary_fan (𝟙 X) (to_pt X ≫ I_1))
+
+-- The mapping cylinder is the colimit of the diagram
+--    X
+--   / \
+--  Y   (X x I)
```

Comment:
You could make the diagrams nicer using the characters `↙ ↘`

Reply:
Thanks!